python/note_input_widget/widget.py

```diff
@@ -456,23 +472,38 @@ def _async_submit_note(self, sg, data):
                 # for older cores, just split on the first space
                 # Sorry Mary Jane Watson!
                 first_name = current_user.get("name").split(" ")[0]
-                
-            title = "%s's Note" % first_name 
+
+            title = "%s's Note" % first_name
         else:
             title = "Unknown user's Note"
-        
+
         if len(note_links) > 0:
             note_names = [x["name"] for x in note_links]
             title += " on %s" % (", ".join(note_names))
 
+        # step 3 - handle project gracefully
+        if project is None:
+            # attempt to pull it from the entity link
+            if entity_project_link is None:
+                # there is no associated project - likely this is a note
+                # on a non-project entity created in the site ctx
+                raise TankError(
```

Comment:
I'm sure this is handled gracefully in higher-level code, right? Do we pop up a dialog or something, or does it just go to the console?